EIPS/eip-770.md

```diff
@@ -0,0 +1,124 @@
+It is the responsibility of the Booker to notify its end user customer of the full resource information and the provider signature (Availability “snapshot” data). This can be used as proof of reservation details in (the extreme) case resource information at reservation time is deleted from off-chain infrastructure.
+
+(Step 4) Provider is validating the reservation request and broadcasts the new status to “RESERVATION_CONFIRMED”.
+
+(Step 5) RES smart contract is updating the reservation status to “RESERVATION_CONFIRMED”
+
+(Step 6.1 - Optional) Booker cancels the reservation by broadcasting a cancellation request. This triggers a submission to the RES smart contract that empties the registries for this resource.
+Depending on the conditions and cancellation date, the BTU escrowed into the RES smart contract are affected back to the user or to the provider
+
+(Step 6.2 - Optional) Provider cancels the reservation by broadcasting a cancellation request. This triggers a submission to the RES smart contract that empties the registries for this resource. RES smart contract is releasing the escrowed amount to booker.
+
+(Step 7.1 & 7.2) Booker is notifying that the resource has been paid (presumably at resource check-out). RES smart contract is releasing the escrowed BTU back to user in addition to a BTU agreed commision.
+
+## Implementation
+
+The implementation involves following functions :
+
+- publishAvailabilities (likely off-chain) : Publishes one or multiple available resources for reservations and for being “searchable”.
+- listAvailabilities (likely off-chain) : List all available resources matching a search criteria
+- requestReservation : Request for a reservation. The function can be used for initial reservation request OR updating
+- getReservationStatus : Read reservation status. Following values are possible : REQUESTED, REJECTED, CONFIRMED
+
+
+```
+contract ERC770 is ERC20 {
+    // Availability structure
+    enum BookingStatus { REQUESTED, REJECTED, CONFIRMED, CANCELLED  }
+    struct availability {
+	address               _contractAdress;
+	uint                     _resourceId ;   // resource id OR bundle id
+	uint                     _type;   // Type of Availability
+	uint                     _minDeposit ; // minimum BTU deposit for booking this resource
+	uint                     _commission ; // commission amount paid to booker in BTU
+	uint                     _freeCancelDateTs; // Limit date for a reservation cancellation
+	uint                     _statDateTs;   //availability start date timestamps
+	uint                     _endDateTs;   //availability end date timestamps
+	BookingStatus    _bookingStatus ; // reservation status
+	string                  _metaDataLink // Link to Meta Data of the bookable resource (desc, image links, etc…)
+   }
+    //Submit one or multiple availability - implementation will be off-chain
```

Comment:
Why does an offchain functionality appear in the interface of an Ethereum contract?

Reply:
We did this for pedagogical reasons (easier to understand).
Also, because sample implementation may implement on-chain functions (for testing without scalability)